### ISSUE TITLE
handling strings as bytes

### ISSUE THREAD

**Issue by geoffwoollard** [asarnow/pyem#19]
I downloaded a .cs file in v2.4.0 (the job was computed with v2.3.2) and tried to convert it to a star file with pyem. The strings in 'ctf/type' and 'blob/path' came out as bytes (b'stringvalue'), so I had to modify pyem to decode this. I'd like to share the fix here:

```python
def simplify_star_ucsf(df, inplace=True):
    df = df if inplace else df.copy()
    if UCSF.IMAGE_ORIGINAL_INDEX in df and UCSF.IMAGE_ORIGINAL_PATH in df:
        df[Relion.IMAGE_ORIGINAL_NAME] = df[UCSF.IMAGE_ORIGINAL_INDEX].map(
            lambda x: "%.6d" % (x + 1)).str.cat(df[UCSF.IMAGE_ORIGINAL_PATH],
                                                sep="@")
# the 3 lines below are the fix
    if UCSF.IMAGE_INDEX in df and UCSF.IMAGE_PATH in df: 
        if type(df[UCSF.IMAGE_PATH][0]) in set([np.bytes_, bytes]) : 
            df[UCSF.IMAGE_PATH] = df[UCSF.IMAGE_PATH].apply(lambda x: x.decode('UTF-8')) 
            
        df[Relion.IMAGE_NAME] = df[UCSF.IMAGE_INDEX].map(
            lambda x: "%.6d" % (x + 1)).str.cat(df[UCSF.IMAGE_PATH], sep="@")
    df.drop([c for c in df.columns if "ucsf" in c or "eman" in c],
            axis=1, inplace=True)
    if "index" in df.columns:
        df.set_index("index", inplace=True)
        df.sort_index(inplace=True, kind="mergesort")
    return df
```

**Comment by asarnow** [comment]
Thanks for the report Geoff. I'm going to put some code to check for this in `pyem.metadata.parse_cryosparc_2_cs()` which is the direct handler for these files.

**Comment by asarnow** [comment]
Upon further investigation, this is caused by using Python 3.

Pyem is only guaranteed to work with Python 2.7, but forward compatibility is a goal. Classifying as an enhancement.

**Comment by asarnow** [comment]
Fixed in 234de70.

This is probably caused by manually choosing a python binary, i.e. `python3 csparc2star.py <args>`. The pyem binaries have shebang lines that choose a particular interpreter, so they should be used directly instead. The correct is simply `csparc2star.py <args>`.
